### ISSUE TITLE
Support for epub3 would be cool

### ISSUE THREAD

**Issue by gdamore** [bytesparadise/libasciidoc#579]
This calls out specifically the work planned for #349 as a base, and then once we have flexible templates, it should be possible to use that as a foundation to create true ebooks.

I'd be willing to invest some effort to assist here.

**Comment by xcoulon** [comment]
hello @gdamore 👋 
(Sorry for the late response here.)
Sure, it would be awesome if you could work on this one, and we can discuss about the design together (as we already started in https://github.com/bytesparadise/libasciidoc/issues/581)

**Comment by gdamore** [comment]
Looking at the XHTML we generate now with -b, and what asciidoctor generates, those are close.  However the asciidoctor epub background generates a quite different (and frankly simpler) version of the HTML for use in ebooks.

For example, there are almost no div elements -- everything <p> elements are provided and the class is attached directly to them.

Unicode entities like the curly quotes are just rendered in UTF-8 directly.

Honestly, the emitted syntax for the EPUB format is a lot nicer to read, and will be simpler to style I think.  We might want to consider a simplified HTML output format.  (The templates we integrated should make this easier.)

**Comment by gdamore** [comment]
Also the epub uses <aside> elements a bit more freely.

**Comment by gdamore** [comment]
I'm working on this now -- what I've found is that the asciidoctor epub3 implementation is strangely buggy -- it almost looks like they aren't really sharing the same parser (they are though).

I'm *not* planning to remain compatible with the asciidoctor-epub3 style sheets.  As I work more and more with the content it generates, I'm finding it less and less pleasant for styling or review, and some of their choices are actually not recommended (like using `<footer>` on a `<div>` -- the `<footer>` element should apply to sections only.  Likewise, they use `<div>` way too much and eschew `<p>`, which is actually not very good for accessibility.

I'm also finding that their choices for some things create some gratuitous visual differences in from the vanilla HTML, which makes styling even harder.

Their use of icons is also ... "interesting" -- I'm leaning towards using SVGs for icons, which from that I can tell work consistently and well, and don't suffer the accessibility flaws associated with font based icons.

Basically, we can do better than asciidoctor-epub3 here, and it's my intent to do so.
